src/rebar_prv_install_deps.erl

```diff
@@ -528,3 +528,7 @@ warn_skip_pkg({Name, Source}, State) ->
         false -> ?WARN(Msg, Args);
         true -> ?ERROR(Msg, Args), ?FAIL
     end.
+
+needs_compile(App) ->
+    rebar_app_info:valid(App)
+        andalso not(rebar_app_info:is_checkout(App)).
```

Comment:
it's faster to check for is_checkout than validity, so we should swap both checks to always fail as fast as possible.